music_assistant/server/providers/soundcloud/__init__.py

```diff
@@ -0,0 +1,338 @@
+    """Provider for Soundcloud."""
+
+    _headers = None
+    _context = None
+    _cookies = None
+    _signature_timestamp = 0
+    _cipher = None
+    _user_id = None
+    _soundcloud = None
+    _me = None
+
+    async def setup(self) -> None:
+        """Set up the Soundcloud provider."""
+        if not self.config.get_value(CONF_CLIENT_ID) or not self.config.get_value(
+            CONF_AUTHORIZATION
+        ):
+            raise LoginFailed("Invalid login credentials")
+
+        client_id = self.config.get_value(CONF_CLIENT_ID)
+        auth_token = self.config.get_value(CONF_AUTHORIZATION)
+
+        async with SoundcloudAsync(auth_token, client_id) as account:
+            await account.login()
+
+        self._soundcloud = account
+        self._me = await account.get_account_details()
+        self._user_id = self._me["id"]
+
+    @property
+    def supported_features(self) -> tuple[ProviderFeature, ...]:
+        """Return the features supported by this Provider."""
+        return SUPPORTED_FEATURES
+
+    @classmethod
+    async def _run_async(cls, call: Callable, *args, **kwargs):
+        return await asyncio.to_thread(call, *args, **kwargs)
+
+    async def search(
+        self, search_query: str, media_types=list[MediaType] | None, limit: int = 10
+    ) -> SearchResults:
+        """Perform search on musicprovider.
+
+        :param search_query: Search query.
+        :param media_types: A list of media_types to include. All types if None.
+        :param limit: Number of items to return in the search (per type).
+        """
+        result = SearchResults()
+        searchtypes = []
+        if MediaType.ARTIST in media_types:
+            searchtypes.append("artist")
+        if MediaType.TRACK in media_types:
+            searchtypes.append("track")
+        if MediaType.PLAYLIST in media_types:
+            searchtypes.append("playlist")
+
+        searchresult = await self._soundcloud.search(search_query, limit)
+
+        for item in searchresult["collection"]:
+            media_type = item["kind"]
+            if media_type == "user":
+                result.artists += [await self._parse_artist(item)]
+            elif media_type == "track":
+                result.tracks += [await self._parse_track(item)]
+            elif media_type == "playlist":
+                result.playlists += [await self._parse_playlist(item)]
+            else:
+                self.logger.debug("Wrong search media_type")
+        return result
+
+    async def get_library_artists(self) -> AsyncGenerator[Artist, None]:
+        """Retrieve all library artists from Soundcloud."""
+        following = await self._soundcloud.get_following(self._user_id)
+        for artist in following["collection"]:
+            try:
+                yield await self._parse_artist(artist)
+            except (KeyError, TypeError, InvalidDataError, IndexError) as error:
+                self.logger.debug("get_library_artists")
```

Comment:
shouldn't this log line be a bit more verbose and merged into one single logline instead of multiple prints ?

Reply:
```suggestion
                self.logger.debug("Parse artist failed: %s", artist, exc_info=error)
```